core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.aop.Pointcut;
+import org.springframework.aop.PointcutAdvisor;
+import org.springframework.aop.framework.AopInfrastructureBean;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.core.Ordered;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.authorization.ReactiveAuthorizationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link MethodInterceptor} which can determine if an {@link Authentication} has access
+ * to the returned object from the {@link MethodInvocation} using the configured
+ * {@link ReactiveAuthorizationManager}.
+ *
+ * @author Evgeniy Cheban
+ */
+public final class AuthorizationManagerAfterReactiveMethodInterceptor implements Ordered, MethodInterceptor,
+		PointcutAdvisor, AopInfrastructureBean, BeanDefinitionRegistryPostProcessor {
```

Comment:
This is necessary so that user can use `AuthorizationManagerAfterReactiveMethodInterceptor` without `@EnableReactiveMethodSecurity` annotation.

For example:
```
@Configuration
class SecurityConfig {

    @Bean
    AuthorizationManagerAfterReactiveMethodInterceptor postAuthorizeInterceptor() {
        return AuthorizationManagerAfterReactiveMethodInterceptor.postAuthorize();
    }

}
```

Reply:
This can be fixed See https://github.com/spring-projects/spring-security/pull/9867/files#diff-d12e230a052788224b32456b63eca596ed0f7213f4463e56690a75b8e0db1dd2R33